tree/ntuple/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -189,14 +335,23 @@ public:
    const std::string &GetProcessorName() const { return fProcessorName; }
 
    /////////////////////////////////////////////////////////////////////////////
-   /// \brief Get the model used by the processor.
-   const ROOT::RNTupleModel &GetModel() const { return *fModel; }
-
-   /////////////////////////////////////////////////////////////////////////////
-   /// \brief Get a reference to the entry used by the processor.
+   /// \brief Register a new field for reading during processing.
+   ///
+   /// \tparam T Type of the field to register.
    ///
-   /// \return A reference to the entry used by the processor.
-   const ROOT::REntry &GetEntry() const { return *fEntry; }
+   /// \param[in] fieldName Name of the field to register.
+   ///
+   /// \return An RNTupleProcessorOptionalPtr, which provides access to the field's value.
+   template <typename T>
+   RNTupleProcessorOptionalPtr<T> RegisterField(const std::string &fieldName)
```

Comment:
Rather than 'registering' a (new) field, isn't this getting an accessor (or handle or proxy or ...)  to the field['s data].   Related ... how/when does it fail if the `fieldName` does not exist in the files?

Reply:
It does both. By calling `RegisterField`, it will be added to the processor's entry so its values will actually be read and an `RNTupleProcessorOptionalPtr` to the field's entry value is returned. Open to alternative naming suggestions!

Thanks for raising the point about how to fail when the field doesn't actually exist, I didn't properly take this into account yet. It lead me to refactor the code in the PR more than expected, but I think this was a necessary step in the right direction. The behavior is now as follows, included in the tests:
https://github.com/root-project/root/blob/bc98934aaaf965750cb24c0174b3a4b19c4b5fca/tree/ntuple/test/ntuple_processor.cxx#L149-L156

Reply:
Maybe rather than `Register` (and its association with creation), what about `RequestField` or simply `GetFieldValuePtr`?